src/shared/modules.mjs

```diff
@@ -854,6 +854,167 @@ const generateEventListenResponse = json => {
     return json
 }
 
+const getRefDefinition = (uri = '', moduleJson = {}) => {
+  if (!uri) {
+    throw "getRefDefinition requires a non-null uri parameter"
+  }
+  if (uri.startsWith('#')) {
+    const path = (uri).split('#').pop()
+
+    let definition
+    if (path) {
+     definition = getPathOr(null, path.slice(1).split('/'), moduleJson)
+    }
+    if (!definition) {
+      throw `getRefDefinition: Path '${uri}' not found in ${moduleJson ? (moduleJson.title || moduleJson.info.title) : moduleJson}.`
+      return null
+    }
+    else {
+      return definition
+    }
+  }
+  else {
+    throw `Cannot call getRefDefinition with ${uri}`
+  }
+}
+
+const getAnyOfSchema = (inType, json, anyOfTypes) => {
+    let outType = localizeDependencies(inType, json)
+    if (outType.schema.anyOf) {
+        if (anyOfTypes.length > 0) {
+        }
+        else {
+            let definition = ''
+            if (inType.schema['$ref'] && (inType.schema['$ref'][0] === '#')) {
+                definition = getRefDefinition(inType.schema['$ref'], json, json['x-schemas'])
+            }
+            else {
+                definition = outType.schema
+            }
+            definition.anyOf.forEach(anyOf => {
+                anyOfTypes.push(anyOf)
+            })
+            outType.schema = anyOfTypes
+        }
+    }
+    return outType
+}
+
+const generateAnyOfSchema = (anyOf, name, summary) => {
+    let anyOfType = []
+    if (!anyOf['$ref']) {
+        anyOf.name = anyOf.title || anyOf.name || ''
```

Comment:
plain JSON schemas shouldn't have `name` attributes. Did you come across a situation where this was needed?

Reply:
No, this is added as an extra check to avoid